cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/RequestAccessEnforcer.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.authorization;
+
+import io.cdap.cdap.api.annotation.Beta;
+import io.cdap.cdap.api.security.AccessException;
+import io.cdap.cdap.proto.element.EntityType;
+import io.cdap.cdap.proto.id.EntityId;
+import io.cdap.cdap.proto.security.Permission;
+import io.cdap.cdap.proto.security.Principal;
+import io.cdap.cdap.proto.security.StandardPermission;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Enforces authorization for the current request {@link Principal} to perform an
+ * {@link Permission} on an {@link EntityId}.
+ * @see AccessEnforcer for enforcer for any principal
+ */
+@Beta
+public interface RequestAccessEnforcer {
```

Comment:
nit: Should we consider renaming to ContextAccessEnforcer? It seems like the DefaultRequestAccessEnforcer uses an AuthenticationContext to determine the context, which may or may not be a request (unless I'm misunderstanding the usage of request)

Reply:
A great idea, let me rename